### ISSUE TITLE
SDKROOT is not set for project build configuration

### ISSUE THREAD

**Issue by philippelatulippe** [yonaskolb/XcodeGen#320]
How can I set the platform of the _project_ (not the target)? When I generate a simple iOS project, xcode seems to use "MacOS Default" instead of "iOS Default" if you look at the Build Settings tab of the Project. This happens because the build variable `SDKROOT` is set on the build configuration of the target, as opposed to that of the project's. Is there something I can set in project.yml so that XcodeGen generates a project with `SDKROOT = iphoneos;`?

Codesigning fails when SDKROOT is left unset on the project. (I use manual signing management)

**Comment by yonaskolb** [comment]
Yes, `SDKROOT` is only set on targets by default. People may have many targets in their project from different platforms, so XcodeGen lets the target take care of that setting. 
You can manually set the `SDKROOT` of the project if you wish like this:

```yml
settings:
  SDKROOT: iphoneos
```

It's strange that you're seeing code signing failing. Let me know if setting the project sdk actually helps that.

**Comment by philippelatulippe** [comment]
Thanks for the details. Setting the SDKROOT in the project settings did fix code signing for me. Without it, Xcode couldn't find the right signing identity.

When I create a project in Xcode, I specify that I am creating an iOS project and the resulting pbxproj does have SDKROOT set on the project. Maybe Xcodegen's project spec could somehow mirror that process?

**Comment by yonaskolb** [comment]
I've opened a PR here that automatically sets the `SDKROOT` build setting on the project, but only if all the targets within it are the same platform
